TopPlaces/TopPlaces/Controllers/DetailsPhotoViewController.m

```diff
@@ -0,0 +1,177 @@
+// Copyright (c) 2016 Lightricks. All rights reserved.
+// Created by Gennadi Iosad.
+
+#import "DetailsPhotoViewController.h"
+
+#import "PhotosHistory.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface DetailsPhotoViewController() <UIScrollViewDelegate>
+
+@property (weak, nonatomic) IBOutlet UIActivityIndicatorView *spinner;
+
+@property (weak, nonatomic) IBOutlet UIScrollView *scrollView;
+
+@property (weak, nonatomic) IBOutlet UILabel *noImageLoadedView;
+
+@property (strong, nonatomic) UIImageView *imageView;
+
+@property (strong, nonatomic) id<CancellableTask> downloadTask;
+
+@end
+
+@implementation DetailsPhotoViewController
+
+#pragma mark -
+#pragma mark UIViewController overrides
+#pragma mark -
+
+- (void)viewDidLoad {
+  [super viewDidLoad];
+  self.scrollView.delegate = self;
+  UITapGestureRecognizer *doubleTapRecognizer =
+      [[UITapGestureRecognizer alloc] initWithTarget:self
+                                              action:@selector(scrollViewDoubleFingerTapped:)];
+  doubleTapRecognizer.numberOfTapsRequired = 2;
+  doubleTapRecognizer.numberOfTouchesRequired = 1;
+  [self.scrollView addGestureRecognizer:doubleTapRecognizer];
+  [self loadImage];
+  NSLog(@"DetailsPhotoViewController::viewDidLoad self %p", self);
+}
+
+- (void)scrollViewDoubleFingerTapped:(UITapGestureRecognizer*)recognizer {
```

Comment:
The name is misleading. A double finger tap is a single tap with 2 fingers <code>scrollViewDoubleTapped:<code> is better.